(GH-2272) Fix complicated quoting in PowerShell Cmdlets (#2300)
puppetlabs/bolt: documentation/running_bolt_commands.md

> 🔩 **Tip:** If a command contains spaces or special shell characters, wrap
> the command in single quotation marks.

### Run a quoted command

> If you need to run a command that uses quotation marks, you must properly escape the quotations. The way you escape the quotations depends on whether you're using Bash or PowerShell.

Reviewer: ```suggestion
If you need to run a command that uses quotation marks, you must properly escape
the quotations. The way you escape the quotations depends on whether you're
using Bash or PowerShell.
```

> If you need to run a command with quotes inside, there are a few considerations to keep in mind. You will need properly escape the quotes according to the whether you are running in `bash` or in `powershell`.

Reviewer 2: ```suggestion
If you need to run a command that uses quotation marks, you must properly escape the quotations. The way you escape the quotations depends on whether you're using Bash or PowerShell.
```

Reviewer 2: Does the Bash stuff apply to all *nix style shells? Just wondering if we should sub Bash for *nix

> In a `bash` shell, you'll need to use backslashes `\` or double the quote you're using in order for the quotes to pass through.

Reviewer 2: ```suggestion
In a Bash shell, use backslashes `\` or double the the quotation marks:
```

_\*nix shell command_

```shell
bolt command run "Get-WMIObject Win32_Service -Filter ""Name like '%mon'""" -t localhost
```

> In a PowerShell shell, use a combination of backslashes `\` and doubling of quotation marks. The example below uses two double quotation marks to quote the value being passed to `Filter`, however the example also uses a backslash so that Bolt's underlying Ruby argument parser accepts the command.

Reviewer: ```suggestion
In a PowerShell shell, use a combination of backslashes `\` and doubling of
quotation marks. The example below uses two double quotation marks to quote the
value being passed to `Filter`, however the example also uses a backslash so
that Bolt's underlying Ruby argument parser accepts the command.
```

> In a `powershell` shell, you'll need to use a combination of escape characters and doubling quotes in order for the quotes to pass through. In the example below, we use two double quotes to quote the value being passed to `Filter`, however we need to pass a backslash in order to have the ruby code below it accept the command.

Reviewer 2: ```suggestion
In a PowerShell shell, use a combination of backslashes `\` and doubling of quotation marks. The example below uses two double quotation marks to quote the value being passed to `Filter`, however the example also uses a backslash so that the Ruby code below the value accepts the command.
```

Reviewer 2: I'm not 100% on "The Ruby code below it". Are we talking about the underlying code that makes up Bolt here?

Author: The ruby argument parser will properly accept the quotes if they are escaped. This is an artifact of passing tokens through two levels of programs. 

Maybe: `so that the Ruby argument parser accepts the command`

Reviewer 2: Okay, I made a small edit. How does that look?

_PowerShell cmdlet_

```powershell

> Invoke-BoltCommand -Command "Get-WMIObject Win32_Service -Filter \""Name like '%mon'\"""

Reviewer: ```suggestion
Invoke-BoltCommand -Command "Get-WMIObject Win32_Service -Filter \""Name like '%mon'\""" -Targets localhost
```
